Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -0,0 +1,1122 @@
+#align localized_module.mk_eq LocalizedModule.mk_eq
+
+@[elab_as_elim]
+theorem induction_on {β : LocalizedModule S M → Prop} (h : ∀ (m : M) (s : S), β (mk m s)) :
+    ∀ x : LocalizedModule S M, β x := by
+  rintro ⟨⟨m, s⟩⟩
+  exact h m s
+#align localized_module.induction_on LocalizedModule.induction_on
+
+@[elab_as_elim]
+theorem induction_on₂ {β : LocalizedModule S M → LocalizedModule S M → Prop}
+    (h : ∀ (m m' : M) (s s' : S), β (mk m s) (mk m' s')) : ∀ x y, β x y := by
+  rintro ⟨⟨m, s⟩⟩ ⟨⟨m', s'⟩⟩
+  exact h m m' s s'
+#align localized_module.induction_on₂ LocalizedModule.induction_on₂
+
+/-- If `f : M × S → α` respects the equivalence relation `LocalizedModule.r`, then
+`f` descents to a map `LocalizedModule M S → α`.
+-/
+def liftOn {α : Type _} (x : LocalizedModule S M) (f : M × S → α)
+    (wd : ∀ (p p' : M × S), p ≈ p' → f p = f p') : α :=
+  Quotient.liftOn x f wd
+#align localized_module.lift_on LocalizedModule.liftOn
+
+theorem liftOn_mk {α : Type _} {f : M × S → α} (wd : ∀ (p p' : M × S), p ≈ p' → f p = f p')
+    (m : M) (s : S) : liftOn (mk m s) f wd = f ⟨m, s⟩ := by convert Quotient.liftOn_mk f wd ⟨m, s⟩
+#align localized_module.lift_on_mk LocalizedModule.liftOn_mk
+
+/-- If `f : M × S → M × S → α` respects the equivalence relation `LocalizedModule.r`, then
+`f` descents to a map `LocalizedModule M S → LocalizedModule M S → α`.
+-/
+def liftOn₂ {α : Type _} (x y : LocalizedModule S M) (f : M × S → M × S → α)
+    (wd : ∀ (p q p' q' : M × S), p ≈ p' → q ≈ q' → f p q = f p' q') : α :=
+  Quotient.liftOn₂ x y f wd
+#align localized_module.lift_on₂ LocalizedModule.liftOn₂
+
+theorem liftOn₂_mk {α : Type _} (f : M × S → M × S → α)
+    (wd : ∀ (p q p' q' : M × S), p ≈ p' → q ≈ q' → f p q = f p' q') (m m' : M)
+    (s s' : S) : liftOn₂ (mk m s) (mk m' s') f wd = f ⟨m, s⟩ ⟨m', s'⟩ := by
+  convert Quotient.liftOn₂_mk f wd _ _
+#align localized_module.lift_on₂_mk LocalizedModule.liftOn₂_mk
+
+instance : Zero (LocalizedModule S M) :=
+  ⟨mk 0 1⟩
+
+@[simp]
+theorem zero_mk (s : S) : mk (0 : M) s = 0 :=
+  mk_eq.mpr ⟨1, by rw [one_smul, smul_zero, smul_zero, one_smul]⟩
+#align localized_module.zero_mk LocalizedModule.zero_mk
+
+instance : Add (LocalizedModule S M)
+    where add p1 p2 :=
+    liftOn₂ p1 p2 (fun x y => mk (y.2 • x.1 + x.2 • y.1) (x.2 * y.2)) <|
+      fun ⟨m1, s1⟩ ⟨m2, s2⟩ ⟨m1', s1'⟩ ⟨m2', s2'⟩ ⟨u1, hu1⟩ ⟨u2, hu2⟩ =>
+          mk_eq.mpr
+            ⟨u1 * u2, by
+              -- Put everything in the same shape, sorting the terms using `simp`
+              have hu1' := congr_arg ((· • ·) (u2 * s2 * s2')) hu1
+              have hu2' := congr_arg ((· • ·) (u1 * s1 * s1')) hu2
+              simp only [smul_add, ← mul_smul, smul_assoc, mul_assoc, mul_comm,
+                mul_left_comm] at hu1' hu2' ⊢
+              rw [hu1', hu2']⟩
+
+theorem mk_add_mk {m1 m2 : M} {s1 s2 : S} :
+    mk m1 s1 + mk m2 s2 = mk (s2 • m1 + s1 • m2) (s1 * s2) :=
+-- Porting note: Simplified thanks to [linter.unreachableTactic].
```

Comment:
```suggestion
```
No need to record simple fixes like this that happened during the port, I think. We'll never do anything with this porting note except delete it!